app/src/main/java/com/yacgroup/yacguide/AscendActivity.kt

```diff
@@ -45,63 +47,60 @@ import java.util.Calendar
 class AscendActivity : AppCompatActivity() {
 
     private lateinit var _db: DatabaseWrapper
-    private var _ascend: Ascend? = null
+    private lateinit var _ascend: Ascend
+    private var _existingAscend: Ascend? = null
     private var _route: Route? = null
-    private var _partnerIds: ArrayList<Int>? = null
-    private var _styleId: Int = 0
-    private var _year: Int = 0
-    private var _month: Int = 0
-    private var _day: Int = 0
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_ascend)
 
         _db = DatabaseWrapper(this)
 
-        _ascend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY, DatabaseWrapper.INVALID_ID))
+        // The database query is null, if this activity is called from the route DescriptionActivity.
+        _existingAscend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY,
+                DatabaseWrapper.INVALID_ID))
+        _ascend = _existingAscend ?: Ascend().apply { routeId = DatabaseWrapper.INVALID_ID }
```

Comment:
I don't understand the body of `apply`? If `_existingAscend` is null, the user has entered via DescriptionActivity and hence, `ROUTE_KEY` must have been set which leads to a valid route id.